addons/web/static/src/core/overlay/overlay_service.js

```diff
@@ -46,6 +46,9 @@ export const overlayService = {
                 props,
                 remove: removeCurrentOverlay,
                 sequence: options.sequence ?? 50,
+                rootId:
+                    props.target?.getRootNode()?.host?.id ||
```

Comment:
I'd move the `props.target?.getRootNode()?.host?.id` part as internal to `popover_service`, as this is specific to the popover kind of overlay, as an option to `overlay.add()`

Reply:
If I move `props.target?.getRootNode()?.host?.id` to the `popover_service`, I think to keep constancy, it would be better to move the `options.context?.root?.el.getRootNode()?.host?.id` part to the `dialog.service` as well so in `overlayService.add()` we have `rootId: options.rootId`.